sdk/include/reactos/shellutils.h

```diff
@@ -94,6 +94,35 @@ inline BOOL _ROS_FAILED_HELPER(HRESULT hr, const char* expr, const char* filenam
 } /* extern "C" */
 #endif /* defined(__cplusplus) */
 
+static inline UINT
+SHELL_ErrorBoxHelper(HWND hwndOwner, UINT Error)
+{
+    WCHAR buf[400];
+    UINT cch, u32_errstr = 2;
+
+    if (!IsWindowVisible(hwndOwner))
+        hwndOwner = NULL;
+    if (Error == ERROR_SUCCESS)
+        Error = ERROR_INTERNAL_ERROR;
+
+    cch = FormatMessageW(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+                         NULL, Error, 0, buf, _countof(buf), NULL);
+    if (!cch)
+    {
+        cch = LoadStringW(LoadLibraryW(L"USER32"), u32_errstr, buf, _countof(buf));
+        wsprintfW(buf + cch, L"\n\n%#x (%d)", Error, Error);
```

Comment:
Looks like a candidate for `StringCchPrintfW` here, though.

Reply:
This is shellutils.h, I don't want to pull in anything. Besides, the string "Error" in any language + two 32-bit formatted numbers is going to fit in `WCHAR buf[400];`.

Reply:
Why not? It's not just about fitting into your buffer. Using unsafe buffer functions can lead to security issues.

Reply:
> It's not just about fitting into your buffer. Using unsafe buffer functions can lead to security issues.

It is only about fitting it into the buffer. We know the format string is safe because we control it here.